packages/react/slot/src/Slot.tsx

```diff
@@ -50,9 +50,8 @@ interface SlotCloneProps {
 
 const SlotClone = React.forwardRef<any, SlotCloneProps>((props, forwardedRef) => {
   const { children, ...slotProps } = props;
-  const child = React.Children.only(children);
-
-  return React.isValidElement(child)
+  const child = React.isValidElement(children) && React.Children.only(children);
```

Comment:
I understand how this fixes the issue when I think about the fact that `React.Children.only` would error, but somehow I find it really confusing to read the diff.

Probably one of these cases where the diff is confusing, but reading the code by itself is ok though.

Reply:
It's behaving as if `React` checks the length of the children internally first before verifying if the child is a react element. Whereas we want to check if it's an element first.

Reply:
Exactly, the confusing part for me when looking at the diff is that we were checking the validity of the element regardless after. So at first glance if you don't really know how `React.Children.Only` behaves, the code kinda looks equivalent, just moved around a bit.

Reply:
I dunno... `false && throw` isn't equivalent to `throw && false`. We needed the former to prevent it falling into the throw.

Reply:
Okay, I've moved this all around a bit. Hopefully it's clearer 🙈